### ISSUE TITLE
useSubscription not working for subscriptions with nested field

### ISSUE THREAD

**Issue by frontendmonster** [apollographql/apollo-client#7415]
**Intended outcome:**

@apollo/client ignore subscriptions that contain object type fields.
Here is an example:

```javascript
const SUBS = gql`
  subscription Foo { 
    bar { 
      field 
      object { nested } 
    }
  }`;

useSubscription(SUBS, { onSubscriptionData() { /* NOT WORKING */ } });
```

**Actual outcome:**

`onSubscriptionData` should get called when the subscription sends data.

**How to reproduce the issue:**

0. Clone the reproduction repo:
`git@github.com:frontendmonster/apollo-client-7415.git`

1. start backend:
`npm start`

2. start frontend:
`npm --prefix client start`

3. open the browser console 

**Versions**
System:
  OS: Linux 5.4 Manjaro Linux
Binaries:
  Node: 14.15.1 - ~/.config/nvm/versions/node/v14.15.1/bin/node
  Yarn: 1.22.10 - /usr/bin/yarn
  npm: 6.14.8 - ~/.config/nvm/versions/node/v14.15.1/bin/npm
Browsers:
  Firefox: 83.0 (It's Brave actually)
npmPackages:
  @apollo/client: 3.3.3 => 3.3.3
  @apollo/link-context: 2.0.0-beta.3 => 2.0.0-beta.3
  @apollo/link-error: 2.0.0-beta.3 => 2.0.0-beta.3
  @apollo/link-ws: 2.0.0-beta.3 => 2.0.0-beta.3

**Comment by brainkim** [comment]
Thank you so much for the reproduction! I’ll look into this and will get back to you soonish (maybe like a week or so). Ping me if I fail to get back to you.

**Comment by frontendmonster** [comment]
Thanks, I am waiting for your response. :pray:

**Comment by brainkim** [comment]
@frontendmonster I pulled down your repository. Looks like the reason the `onSubscriptionData` wasn’t being called was that there was an error (`Counter.guess` was required by your schema but you didn’t provide it from the server). It’s often helpful to run your queries in a separate interface like [Apollo Client Devtools](https://github.com/apollographql/apollo-client-devtools), the [GraphiQL IDE](https://www.apollographql.com/docs/apollo-server/v1/graphiql/), or [Apollo Studio](https://www.apollographql.com/docs/studio/) just to make sure everything is okay.

Other than that, I would make sure to check for errors by reading the return value of the hook. Closing for housekeeping purposes, but thanks for reaching out, and feel free to ping me if you have more thoughts!
